### ISSUE TITLE
Record constraint polymorphic function

### ISSUE THREAD

**Issue by haitlahcen** [dhall-lang/dhall-haskell#968]
Hello,

I found myself trying to extend a record in a function accepting a type, currently, I don't really see how I can achieve this:

```
let f = \(T : Type) -> \(x : T) -> let A = { a : Text, b : Text } in { a = "", b = "" } /\ x : A //\\ X

in f { c : Text } { c = "" }
```

I could have a nested value inside the first record, but having a flat record is impossible right now ?

**Comment by Gabriel439** [comment]
@haitlahcen: It's currently not possible within the language.  The reason why is that Dhall does not distinguish between record types and non-record types at the kind-level, so there's no way for the type-checker to know that the `T` in `\(T : Type) -> ...` is a record type.

**Comment by haitlahcen** [comment]
@Gabriel439  would it be usefull to differentiate them ?

**Comment by Gabriel439** [comment]
@haitlahcen: Yes, but it just takes time to design how to do so and standardize it.

**Comment by Profpatsch** [comment]
I’d like this to be possible as well. It looked strange to me that `{ a = Integer } : { a : Type }` but `{ a : Integer } : Type`.

**Comment by Gabriel439** [comment]
@Profpatsch: Note that fixing this would most likely not introduce a kind-level record.  It would probably just introduce a new kind like `Record Type`:

```haskell
{ a = Integer } : { a : Type } : Record Type
```

**Comment by sjakobi** [comment]
I'll transfer this issue to the dhall-lang repo since it is about a change in the language, not in the Haskell implementation.
